test/TritonGPU/combine.mlir

```diff
@@ -363,7 +363,7 @@ tt.func @loop_if(%arg0: !tt.ptr<f32>, %arg1: i32, %arg2: !tt.ptr<f32>, %arg3: i3
   %9 = triton_gpu.convert_layout %8 : tensor<64x64xi32, #blocked2> -> tensor<64x64xi32, #blocked1>
   %10 = tt.addptr %7, %9 : tensor<64x64x!tt.ptr<f32>, #blocked1>, tensor<64x64xi32, #blocked1>
   %11:2 = scf.for %arg5 = %c0 to %c32 step %c1 iter_args(%arg6 = %cst_1, %arg7 = %10) -> (tensor<64x64xf32, #blocked1>, tensor<64x64x!tt.ptr<f32>, #blocked1>) {
-    %33 = arith.cmpi "sgt", %i0, %i0 : i32
+    %33 = arith.cmpi "sgt", %arg5, %c0 : index
```

Comment:
This seemingly unrelated change is because `i0 > i0` is a constant and the `IfOp` canonicalization patterns added in this PR was removing it.